policies/tickets.rego

```diff
@@ -2,47 +2,63 @@ package tickets
 
 import rego.v1
 
-import data.roles
+roles := data.roles
+
+response.allow := allow
+response.reason := reason_admin
+response.conditions := conditions
 
 default allow := false
 allow if allowed(input.user, input.tenant, input.action)
 
 allowed(user, tenant, _) if "admin" in roles[tenant][user]
+
 allowed(user, tenant, action) if {
-    "reader" in roles[tenant][user]
-    action in {"get", "list"}
+	action in {"get", "list"}
+	some role in roles[tenant][user]
+	read_allowed(role)
 }
-allowed(user, tenant, "resolve") if "resolver" in roles[tenant][user]
+allowed(user, tenant, "resolve") if user_is_resolver(user, tenant)
+
+read_allowed("reader")
+read_allowed("resolver")
+
+user_is_resolver(user, tenant) if "resolver" in roles[tenant][user]
+
+## CONDITIONS ##
+
+#conditions["tickets.resolved"] := false if user_is_resolver(input.user, input.tenant)
+#conditions["tickets.id"].ne := 2
+
+conditions.or contains {"tickets.resolved": false} if user_is_resolver(input.user, input.tenant)
+conditions.or contains {"tickets.id": {"ne": 2}}
+conditions["customers.name"] := "Globex"
```

Comment:
At this point, these are still for playing with the structure and its interactions with prisma. Will need to update to meet actual requirements.